cocotb/__init__.py

```diff
@@ -56,20 +56,9 @@
 
 # GPI logging instance
 if "COCOTB_SIM" in os.environ:
-    import simulator
-    logging.basicConfig()
-    logging.setLoggerClass(SimBaseLog)
-    log = SimLog('cocotb')
-    level = os.getenv("COCOTB_LOG_LEVEL", "INFO")
-    try:
-        _default_log = getattr(logging, level)
-    except AttributeError as e:
-        log.error("Unable to set loging level to %s" % level)
-        _default_log = logging.INFO
-    log.setLevel(_default_log)
-    loggpi = SimLog('cocotb.gpi')
-    # Notify GPI of log level
-    simulator.log_level(_default_log)
+    cocotb.log.default_config()
+
+    log = logging.getLogger(__name__)
 
     # If stdout/stderr are not TTYs, Python may not have opened them with line
     # buffering. In that case, try to reopen them with line buffering
```

Comment:
After thinking some more, these lines should now be _above_ the logging config.

Before, what happened was:

* The `cocotb` log object was configured with stdout
* `stdout` was fixed
* The `cocotb.scheduler` and all other nested log objects were configured

As currently written, the behavior is

* All log objects are configured with stdout
* `stdout` is fixed

This probably ought to be
* `stdout` is fixed
* All log objects are configured with stdout